### PR TITLE
Results are now cached, format limits to other networks are respected

### PR DESCRIPTION
**Pull request by annando**
This belongs to PR https://github.com/friendica/friendica/pull/4636

Additionally to the improvements for exports to other networks, there is now a caching of requests to avoid restrictions by the OSM servers.

### CODE REVIEW

File: openstreetmap/openstreetmap.php [hunk 1]
```diff
@@ -12,13 +12,20 @@
 use Friendica\Core\Cache;
 use Friendica\Core\Config;
 use Friendica\Core\L10n;
+use Friendica\Core\System;
 use Friendica\Util\Network;
 
+const OSM_TMS = 'http://www.openstreetmap.org';
+const OSM_NOM =	'http://nominatim.openstreetmap.org/search.php';
```
Comment (tobiasd) [inline openstreetmap/openstreetmap.php]: can you remove this tab?

File: openstreetmap/openstreetmap.php [hunk 2]
```diff
@@ -12,13 +12,20 @@
...
+const OSM_ZOOM = 16;
+const OSM_MARKER = 0;
+
 function openstreetmap_install()
 {
 	Addon::registerHook('render_location', 'addon/openstreetmap/openstreetmap.php', 'openstreetmap_location');
 	Addon::registerHook('generate_map', 'addon/openstreetmap/openstreetmap.php', 'openstreetmap_generate_map');
 	Addon::registerHook('generate_named_map', 'addon/openstreetmap/openstreetmap.php', 'openstreetmap_generate_named_map');
+	Addon::registerHook('Map::getCoordinates', 'addon/openstreetmap/openstreetmap.php', 'openstreetmap_get_coordinates');
```
Comment (tobiasd) [inline openstreetmap/openstreetmap.php]: has the notation of the hooks changed? If so, why not convert all called hooks to the new notation?

Comment (annando) [reply openstreetmap/openstreetmap.php]: The old ones had been named after the functions they came from. We renamed the functions, but the hook names stayed the same. This hook is new, so I named it after the function again.

Comment (tobiasd) [reply openstreetmap/openstreetmap.php]: As it is new, could you then please add it to the addon/hook documentation.

**Comment by tobiasd** [comment]
Should these changes be applied to the googlemaps addpn as well?

**Comment by annando** [comment]
I hadn't realized that we are having one ;-)

Yeah, I can submit an additional PR for that. But this is not blocking for this PR or the core PR.

**Comment by tobiasd** [comment]
Ok, thanks for checking :-)

**Pull request merged**